MaxText/layers/gpt3.py

```diff
@@ -180,85 +179,122 @@ class Gpt3MultiHeadAttention(nn.Module):
-  attention_kernel: str
-  dtype: DType = jnp.float32
-  weight_dtype: DType = jnp.float32
-  dropout_rate: float = 0.0
-  kernel_init: NdInitializer = nd_dense_init(1.0, "fan_in", "normal")
-  float32_qk_product: bool = False  # computes logits in float32 for stability.
-  float32_logits: bool = True  # cast logits in float32 for stability.
-  fused_qkv: bool = True
-  quant: Optional[Quant] = None
-  kv_quant: Optional[KVQuant] = None
-  use_bias: bool = True
-
-  input_axis_names: AxisNames = (BATCH, LENGTH, EMBED)
-  query_axis_names: AxisNames = (BATCH, LENGTH, HEAD, D_KV)
-  key_axis_names: AxisNames = (BATCH, LENGTH, HEAD, D_KV)
-  value_axis_names: AxisNames = (BATCH, LENGTH, HEAD, D_KV)
-  out_axis_names: AxisNames = (BATCH, LENGTH, HEAD, D_KV)
-
-  def qkv_projection(self, inputs: Array, proj_name: str):
-    """Fused QKV projection"""
+  def __init__(
+      self,
+      config: Config,
+      num_heads: int,
+      feature_dim: Array,
+      head_dim: int,
+      max_target_length: int,
+      max_prefill_predict_length: int,
+      mesh: Mesh,
+      attention_kernel: str,
+      dtype: DType = jnp.float32,
+      weight_dtype: DType = jnp.float32,
+      dropout_rate: float = 0.0,
+      kernel_init: NdInitializer = nd_dense_init(1.0, "fan_in", "normal"),
+      float32_qk_product: bool = False,  # computes logits in float32 for stability.
+      float32_logits: bool = True,  # cast logits in float32 for stability.
+      fused_qkv: bool = True,
+      quant: Optional[Quant] = None,
+      kv_quant: Optional[KVQuant] = None,
+      use_bias: bool = True,
+      input_axis_names: AxisNames = (BATCH, LENGTH, EMBED),
+      query_axis_names: AxisNames = (BATCH, LENGTH, HEAD, D_KV),
+      key_axis_names: AxisNames = (BATCH, LENGTH, HEAD, D_KV),
+      value_axis_names: AxisNames = (BATCH, LENGTH, HEAD, D_KV),
+      out_axis_names: AxisNames = (BATCH, LENGTH, HEAD, D_KV),
+      rngs: Optional[nnx.Rngs] = None,
+      **kwargs: Any,
+  ):
+    self.config = config
+    self.num_heads = num_heads
+    self.head_dim = head_dim
+    self.max_target_length = max_target_length
+    self.max_prefill_predict_length = max_prefill_predict_length
+    self.mesh = mesh
+    self.attention_kernel = attention_kernel
+    self.dtype = dtype
+    self.weight_dtype = weight_dtype
+    self.dropout_rate = dropout_rate
+    self.kernel_init = kernel_init
+    self.float32_qk_product = float32_qk_product
+    self.float32_logits = float32_logits
+    self.fused_qkv = fused_qkv
+    self.quant = quant
+    self.kv_quant = kv_quant
+    self.use_bias = use_bias
+    self.input_axis_names = input_axis_names
+    self.query_axis_names = query_axis_names
+    self.key_axis_names = key_axis_names
+    self.value_axis_names = value_axis_names
+    self.out_axis_names = out_axis_names
+    self.rngs = rngs if rngs is not None else kwargs.get("rngs", nnx.Rngs(0))
+    if self.fused_qkv:
+      self.qkv_projection_layer = self.create_projection_layer(
```

Comment:
The attribute name `self.<attribute_name>` needs to match the previous `name` value passed into the Linen module. This is for checkpointing compatibility purposes (so the old Linen checkpoints will still work with the new NNX ones). Could you update these to make them match? I believe this one should be `self.qkv_proj` for example